File: shared-web/src/types/event.ts
import type { BarLocation } from './bar'

export interface CafeEvent {
  id: number
  bar: BarLocation
  title: string
  /** Event date as "YYYY-MM-DD". */
  date: string
  /** Start time as "HH:mm", or null when not set. */
  startTime: string | null
  /** Free-text price, including "TBD" or "Free". */
  price: string | null
  description: string | null
  imageId: number | null
  imageUrl: string | null
  imageAlt: string | null
  /** External sign-up/subscribe link, or null. */
  subscribeLink: string | null
  published: boolean
}

File: backend/src/main/java/cafe/community/backend/model/Event.java
package cafe.community.backend.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;

/**
 * A bar event. Always scoped to one bar; null bar is not used for events.
 * {@code published = false} hides the event from the public site without deleting it.
 */
@Data
@Entity
@Table(name = "event")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private BarLocation bar;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDate date;

    @Column(name = "start_time")
    private LocalTime startTime;

    /** Free-text price, e.g. "Free", "€5", "TBD". */
    @Column(length = 100)
    private String price;

    /** External sign-up or info link. */
    @Column(name = "subscribe_link", length = 512)
    private String subscribeLink;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "image_id")
    private MediaAsset image;

    @Column(nullable = false)
    private boolean published = true;
}
